ecl/hql/hqlopt.cpp

```diff
@@ -1166,51 +1166,61 @@ IHqlExpression * CTreeOptimizer::getHoistedFilter(IHqlExpression * transformed,
             }
         }
 
-        if (!matched && !expandMonitor.isComplex())
+        try
```

Comment:
It's a little more efficient to have the try block inside the if block below.

Reply:
try blocks have no cost with modern implementations - so there is no difference in efficiency.